File: mall-manager-web/src/main/webapp/js/service/itemCatalogService.js
/**
 * 商品分类服务层
 * Created by Wwl on 2019/3/4
 */
app.service('itemCatalogService', function($http){

    //增加
    this.add = (entity) => $http.post('../itemCatalog/add.do',entity);

    //修改
    this.update = (entity) => $http.post('../itemCatalog/update.do',entity);

    //删除
    this.delete = (ids) => $http.get('../itemCatalog/delete.do?ids='+ids);

    //查询实体
    this.findOne = (id) => $http.get('../itemCatalog/findOne.do?id='+id);

    //查询列表
    this.findList = () => $http.get('../itemCatalog/list.do');

    //分页查询
    this.findPage = (page,size) => $http.get('../itemCatalog/listByPage.do?page=' + page + '&size=' + size);

    //搜索
    this.search = (page,size,searchEntity) => $http.post('../itemCatalog/search.do?page='+page+'&size='+size,searchEntity);

    //根据pId获得列表
    this.findByParentId = (parentId) => $http.get('../itemCatalog/findByParentId.do?parentId='+parentId);

});

File: mall-goods-interface/src/main/java/com/mall/goods/service/ItemCatalogService.java
package com.mall.goods.service;

import com.mall.pojo.ItemCatalog;
import common.pojo.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 商品目录/分类服务接口
 * @author Wwl
 *
 */
public interface ItemCatalogService {

	/**
	 * 增加
	 * @param catalog
	 */
	public void add(ItemCatalog catalog);

	/**
	 * 修改
	 * @param catalog
	 */
	public void update(ItemCatalog catalog);

	/**
	 * 根据Id获取实体
	 * @param id
	 * */
	public ItemCatalog findOne(Long id);

	/**
	 * 查询所有
	 * @return
	 */
	public List<ItemCatalog> findAll();
	
	/**
	 * 查询所有-分页
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);

	/**
	 * 根据条件查询，分页
	 * @param catalog
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(ItemCatalog catalog, int pageNum, int pageSize);

	/**
	 * 删除
	 * @return
	 */
	public void delete(Long[] ids);

	/**
	 * 根据父ID查询列表
	 * @param id
	 * @return
	 */
	public List<ItemCatalog> findByParentId(Long id);

}
